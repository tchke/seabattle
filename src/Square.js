import React, { Component } from 'react';
import classNames from 'classnames';

export default class Square extends Component {
    constructor(props){
        super(props);
        this.state = {};
    }

    handleClick = (e) => {
        this.setState({exposed: true});
    };

    render() {
        const { has } = this.props;
        const name = has ? 'red' : 'grey';
        const className = classNames('square', name);
        return (
            <button className={className} onClick={this.handleClick}>
                {/* TODO */}
            </button>
        );
    }
}
