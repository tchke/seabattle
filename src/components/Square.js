import React, { Component } from 'react';
import classNames from 'classnames';

export default class Square extends Component {
    handleClick = () => {
        const { onClick, i, j } = this.props;
        console.log(`clicked on ${i} ${j}`);
        onClick(i,j);
    }

    render() {
        const { has, open } = this.props;
        const name = open ? has ? 'red' : 'white' : 'grey';
        const className = classNames('square', name);
        return (
            <button className={className} onClick={this.handleClick}>
                {/* TODO */}
            </button>
        );
    }

}
