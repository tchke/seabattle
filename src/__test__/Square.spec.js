import React from 'react';
import Square from '../components/Square';
import renderer from 'react-test-renderer';

it('empty square should be rendered correctly', () => {
    const tree = renderer
        .create(<Square></Square>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('open empty square should be rendered correctly', () => {
    const tree = renderer
        .create(<Square open={true}></Square>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('occupied square should be rendered correctly', () => {
    const tree = renderer
        .create(<Square has={true}></Square>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('open occupied square should be rendered correctly', () => {
    const tree = renderer
        .create(<Square has={true} open={true}></Square>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

