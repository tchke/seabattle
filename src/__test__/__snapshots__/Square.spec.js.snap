// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty square should be rendered correctly 1`] = `
<button
  className="square grey"
  onClick={[Function]}
/>
`;

exports[`occupied square should be rendered correctly 1`] = `
<button
  className="square grey"
  onClick={[Function]}
/>
`;

exports[`open empty square should be rendered correctly 1`] = `
<button
  className="square white"
  onClick={[Function]}
/>
`;

exports[`open occupied square should be rendered correctly 1`] = `
<button
  className="square red"
  onClick={[Function]}
/>
`;
